UnitySDK/Assets/ML-Agents/Scripts/Policy/BehaviorParameters.cs

```diff
@@ -0,0 +1,60 @@
+using Barracuda;
+using System;
+using UnityEngine;
+
+namespace MLAgents
+{
+
+    /// <summary>
+    /// The Factory to generate policies. 
+    /// </summary>
+    public class BehaviorParameters : MonoBehaviour
+    {
+
+        [HideInInspector]
+        [SerializeField]
+        private BrainParameters m_BrainParameters = new BrainParameters();
+        [HideInInspector] [SerializeField] private NNModel m_Model;
+        [HideInInspector] [SerializeField] private InferenceDevice m_InferenceDevice;
+        [HideInInspector] [SerializeField] private bool m_UseHeuristic;
+        [HideInInspector] [SerializeField] private string m_BehaviorName = "My Behavior";
+
+        [HideInInspector]
+        public BrainParameters brainParameters
+        {
+            get { return m_BrainParameters; }
+        }
+
+        [HideInInspector]
+        public string behaviorName
+        {
+            get { return m_BehaviorName; }
+        }
+
+        public IPolicy GeneratePolicy(Func<float[]> heuristic)
+        {
+            if (m_Model == null || m_UseHeuristic)
```

Comment:
Might need to talk about the order here. If you're starting from scratch and don't have a model, you probably want to train (hopefully the communicator is on). As it's written now, you'll get a HeuristicPolicy which probably will raise an exception.

Reply:
Maybe just `if(m_UseHeuristic)` at the start, and also return a HeuristicPolicy if no model and no communicator?

Reply:
In my view, I wanted the Heuristic to have priority over training. But we should discuss that.

Reply:
This sounds good to me